rpc/codec/conn.go

```diff
@@ -9,94 +9,50 @@ import (
 	"encoding/binary"
 	"io"
 	"net"
-	"sync"
-
-	"code.google.com/p/snappy-go/snappy"
 
+	wire "github.com/cockroachdb/cockroach/rpc/codec/wire.pb"
 	"github.com/gogo/protobuf/proto"
 )
 
-type frameBuf struct {
-	size [binary.MaxVarintLen64]byte
+const enableSnappy = true
+
+type decompressFunc func(src []byte, m proto.Message) error
+
+var decompressors = [...]decompressFunc{
```

Comment:
`[...]` creates an array, not a map. The "keys" below are actually `goto` labels, which have no effect here. We should either remove the labels and rely on the fact that the array matches the order of the enum, or make this a map which would appear to be the original intention.